src/tests/e2e/estimatingMonthlyCostUserStoryE2E.spec.ts

```diff
@@ -0,0 +1,158 @@
+        await calculatorPage.chooseMachineType("n1-standard-2");
+        await page.waitForTimeout(2000);
+        let firstUITotal = await calculatorPage.getEstimatedCost();
+
+        await calculatorPage.clickAddEstimate();
+        await calculatorPage.selectComputeEngine();
+        await calculatorPage.addInstances(4);
+        await page.waitForTimeout(2000);
+        let secondUITotal = await calculatorPage.getEstimatedCost();
+
+        expect(firstUITotal).not.toBe(secondUITotal);
+
+        firstUITotal = await calculatorPage.getDisplayedCostsOnCostDetails(1);
+        secondUITotal = await calculatorPage.getDisplayedCostsOnCostDetails(2);
+
+        expect(firstUITotal).not.toBe(secondUITotal);
+    });
+
+    test('should realtime update the cost estimation on instance value change', async ({page}) => {
+        await calculatorPage.clickAddEstimate();
+        await calculatorPage.selectComputeEngine();
+        await page.waitForTimeout(1000);
+        await calculatorPage.addInstances(1);
+        await page.waitForTimeout(2000);
+        let firstUITotal = await calculatorPage.getEstimatedCost();
+
+        await calculatorPage.addInstances(2);
+        await page.waitForTimeout(2000);
+        let secondUITotal = await calculatorPage.getEstimatedCost();
+
+        expect(firstUITotal).not.toBe(secondUITotal);
+    });
+
+    test('should estimate cost with minimum values', async ({page}) => {
+        await calculatorPage.clickAddEstimate();
+        await calculatorPage.selectComputeEngine();
+
+        await calculatorPage.setInstanceTime("5");
+        await calculatorPage.chooseSpotProvisionModel();
+        await calculatorPage.chooseMachineType("f1-micro");
+        await calculatorPage.chooseStandardPersistentDisk();
+        await calculatorPage.setBootDiskSize("1");
+        await page.waitForTimeout(2000);
+        let uiTotal = await calculatorPage.getEstimatedCost();
+
+        expect(uiTotal).toBe(0.02);
+    });
+
+    test('should estimate cost with maximum values', async ({page}) => {
+        await calculatorPage.clickAddEstimate();
+        await calculatorPage.selectComputeEngine();
+
+        await calculatorPage.setInstanceTime("36500000");
+        await calculatorPage.chooseMachineType("n1-highcpu-96");
+        await calculatorPage.chooseMachineFamily("storage-optimized");
+        await calculatorPage.setBootDiskSize("65536");
+        await page.waitForTimeout(2000);
+        let uiTotal = await calculatorPage.getEstimatedCost();
+
+        expect(uiTotal).toBe(736644346);
+    });
+
+    test('should not estimate cost specifying 0 number to instances', async ({page}) => {
+        await calculatorPage.clickAddEstimate();
+        await calculatorPage.selectComputeEngine();
+
+        await calculatorPage.setInstanceTime("0");
+        await page.waitForTimeout(2000);
+        const uiTotal = await calculatorPage.getEstimatedCost();
+
+        expect(uiTotal).toBeNaN();
```

Comment:
toBeNan is not the best option here, I would recommend to check that total cost is not displayed at all